Move WebCodecs VideoFrame serialisation out of agent cluster tests to a separate file

youennf opened the pull request:
#### ab6979eff99f1aa2cfe73ad7cf3c66fcad26c02c
<pre>
Move WebCodecs VideoFrame serialisation out of agent cluster tests to a separate file
<a href="https://bugs.webkit.org/show_bug.cgi?id=260954">https://bugs.webkit.org/show_bug.cgi?id=260954</a>
rdar://114750260

Reviewed by Tim Nguyen.

As per <a href="https://github.com/web-platform-tests/wpt/issues/41635">https://github.com/web-platform-tests/wpt/issues/41635</a>, we split tests in two so as to move the crossAgentCluster tests out of Interop 2023.
Move tests from videoFrame-serialization.crossAgentCluster.https.html to videoFrame-serialization.https.html.
For all tests, we are making sure to close the frame explicityl at the end of the test so that the frame does not get gced unclosed.
The remaining tests should be changed to accept transfer/serialization but that can be done as a follow-up.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.crossAgentCluster.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.crossAgentCluster.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.https.html: Added.

Canonical link: <a href="https://commits.webkit.org/267545@main">https://commits.webkit.org/267545@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6fa5adee88423d8c4eec774ffcc7b450bb27e8fe

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16921 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17244 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17702 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18704 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15840 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20476 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17382 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18085 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17120 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17476 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19513 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14719 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15346 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22066 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15703 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15513 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19824 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16118 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/13656 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15281 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/15186 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4051 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19645 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15948 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/resources/import-expectations.json:
```diff
@@ -444,6 +444,7 @@
     "web-platform-tests/webaudio": "import",
     "web-platform-tests/webauthn": "skip",
     "web-platform-tests/webcodecs": "import",
+    "web-platform-tests/webcodecs/videoFrame-serialization.crossAgentCluster.https.html": "import",
```
Can we remove this line?

youennf replied:
done

youennf committed:
Move WebCodecs VideoFrame serialisation out of agent cluster tests to a separate file
https://bugs.webkit.org/show_bug.cgi?id=260954
rdar://114750260

Reviewed by Tim Nguyen.

As per https://github.com/web-platform-tests/wpt/issues/41635, we split tests in two so as to move the crossAgentCluster tests out of Interop 2023.
Move tests from videoFrame-serialization.crossAgentCluster.https.html to videoFrame-serialization.https.html.
For all tests, we are making sure to close the frame explicityl at the end of the test so that the frame does not get gced unclosed.
The remaining tests should be changed to accept transfer/serialization but that can be done as a follow-up.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.crossAgentCluster.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.crossAgentCluster.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webcodecs/videoFrame-serialization.https.html: Added.

Canonical link: https://commits.webkit.org/267545@main
